File: ios/Plugin/SignInWithApplePlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(SignInWithApplePlugin)
public class SignInWithApplePlugin: CAPPlugin {
    private var implementation: SignInWithApple?
    
    
    override public func load() {
        implementation = SignInWithApple(viewController: self.bridge?.viewController)
    }

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation?.echo(value) ?? ""
        ])
    }

    @objc func getCanShowButton(_ call: CAPPluginCall) {
        call.resolve([
            "value": implementation?.canShowButton ?? false
        ])
    }

    @objc func login(_ call: CAPPluginCall) {
        implementation?.login { (result) in
            var r: [String: Any] = [:]
            if let info = result as? [String : Any] {
                r = info
            }
            call.resolve(r)
        }
    }

    @objc func checkStateWithUserID(_ call: CAPPluginCall) {
        guard let id = call.options["userId"] as? String else {return}
        implementation?.checkState(withUserID: id) { (result) in
            var r: [String: Any] = [:]
            if let info = result as? [String : Any] {
                r = info
            }
            call.resolve(r)
        }
    }
}

File: android/src/main/java/capacitor/plugin/sign/in/with/apple/v3/SignInWithApplePlugin.java
package capacitor.plugin.sign.in.with.apple.v3;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "SignInWithApple")
public class SignInWithApplePlugin extends Plugin {

    private SignInWithApple implementation = new SignInWithApple();

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }

    @PluginMethod
    public void getCanShowButton(PluginCall call) {
        JSObject ret = new JSObject();
        ret.put("value", "getCanShowButton method only works on ios ");
        call.resolve(ret);
    }

    @PluginMethod
    public void login(PluginCall call) {
        JSObject ret = new JSObject();
        ret.put("value", "login method only works on ios ");
        call.resolve(ret);
    }

    @PluginMethod
    public void checkStateWithUserID(PluginCall call) {
        JSObject ret = new JSObject();
        ret.put("value", "checkStateWithUserID method only works on ios ");
        call.resolve(ret);
    }
}
